src/framework/ui/view/abstractmenumodel.h

```diff
@@ -84,24 +76,24 @@ class AbstractMenuModel : public QAbstractListModel, public async::Asyncable
     static const int INVALID_ITEM_INDEX;
     int itemIndex(const QString& itemId) const;
 
-    MenuItem& item(int index);
+    MenuItem* item(int index);
 
-    MenuItem& findItem(const QString& itemId);
-    MenuItem& findItem(const actions::ActionCode& actionCode);
-    MenuItem& findMenu(const QString& menuId);
+    MenuItem* findItem(const QString& itemId);
```

Comment:
I suggest returning constant references (instead of raw pointers) where possible. This will help to avoid countless checks for null pointers